midp/gfx.js

```diff
@@ -1039,82 +996,24 @@ var currentlyFocusedTextEditor;
     }
 
     function clipRect(g, x, y, width, height) {
-        var translatedX1, translatedY1;
-        var translatedX2, translatedY2;
-
         if (width <= 0 || height <= 0) {
             g.clipX1 = g.clipY1 = g.clipX2 = g.clipY2 = 0;
-            g.clipped = true;
-            return;
-        }
-
-        // Translate the given coordinates
-        translatedX1 = x + g.transX;
-        translatedY1 = y + g.transY;
-
-        // Detect overflow
-        if (translatedX1 < 0) {
-            translatedX1 = (x < 0 || g.transX < 0) ? 0 : g.maxWidth;
-        }
-        if (translatedY1 < 0) {
-            translatedY1 = (y < 0 || g.transY < 0) ? 0 : g.maxHeight;
-        }
```

Comment:
This section of code doesn't make sense in JavaScript; we're not going to "overflow" a JavaScript `Number` and, if we do, we're not going to get a negative number - we'll get `Number.POSITIVE_INFINITY`